### PR TITLE
Add WindowsAutopilotDeploymentProfile

### PR DESCRIPTION
**Pull request by kenchan0130**
This adds initial support for the WindowsAutopilotDeploymentProfile resource.

These are APIs related to Intune, not Azure AD.
I intend to gradually add more intune APIs in the future.

Graph API Reference:

https://learn.microsoft.com/en-us/graph/api/resources/intune-shared-windowsautopilotdeploymentprofile https://learn.microsoft.com/en-us/graph/api/resources/intune-enrollment-azureadwindowsautopilotdeploymentprofile https://learn.microsoft.com/en-us/graph/api/resources/intune-enrollment-activedirectorywindowsautopilotdeploymentprofile

### CODE REVIEW

**Comment by manicminer** [comment]
Thank you for the PR @kenchan0130. We recently merged an update that removes support for specifying the tenant ID in clients - would you be able to rebase from main and also resolve the conflict in `odata/types.go` and then I can look at merging this? Thanks!

**Comment by kenchan0130** [comment]
I will fix this PR when https://github.com/hashicorp/go-azure-sdk/pull/342 changes are released.

File: msgraph/windows_autopilot_deployment_profiles.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+// Create creates a new WindowsAutopilotDeploymentProfile.
+func (c *WindowsAutopilotDeploymentProfilesClient) Create(ctx context.Context, profile WindowsAutopilotDeploymentProfile) (*WindowsAutopilotDeploymentProfile, int, error) {
+	var status int
+	body, err := json.Marshal(profile)
+	if err != nil {
+		return nil, status, fmt.Errorf("json.Marshal(): %v", err)
+	}
+	hoge := string(body)
+	fmt.Printf(hoge)
```
Comment (kenchan0130) [inline msgraph/windows_autopilot_deployment_profiles.go:63]: These are my debug code.
I will remove them.

Comment (kenchan0130) [reply msgraph/windows_autopilot_deployment_profiles.go]: fixed by https://github.com/manicminer/hamilton/pull/228/commits/99cc5c05c3b1d582d036cd81fd5bd482dfc64014

**Comment by kenchan0130** [comment]
@manicminer I have made revisions based on your feedback. Please review the changes once more.

**Comment by kenchan0130** [comment]
@manicminer 
Thank you for your review.

> Application is not authorized to perform this operation. Application must have one of the following scopes: DeviceManagementServiceConfiguration.ReadWrite.All, DeviceManagementServiceConfig.ReadWrite.All

From the errors in the test at CI, it seems that the Application used in the tests does not have enough permissions.
I think that if you grant permissions, the tests that are failing will probably succeed.

**Pull request merged**